pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -1351,56 +1408,145 @@ func (vs *VSphere) NodeExists(nodeName k8stypes.NodeName) (bool, error) {
 	return false, nil
 }
 
-func (vs *VSphere) createDummyVM(ctx context.Context, datacenter *object.Datacenter, datastore *object.Datastore) (*object.VirtualMachine, error) {
+// A background routine which will be responsible for deleting stale dummy VM's.
+func (vs *VSphere) cleanUpDummyVMs(dummyVMPrefix string) {
+	// Create context
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	for {
+		// Ensure client is logged in and session is valid
+		vSphereLogin(ctx, vs)
+
+		// Create a new finder
+		f := find.NewFinder(vs.client.Client, true)
+
+		// Fetch and set data center
+		dc, err := f.Datacenter(ctx, vs.cfg.Global.Datacenter)
+		f.SetDatacenter(dc)
+
+		// Get the folder reference for global working directory where the dummy VM needs to be created.
+		vmFolder, err := getFolder(ctx, vs.client, vs.cfg.Global.Datacenter, vs.cfg.Global.WorkingDir)
+		if err == nil {
+			dummyVMRefList := getDummyVMList(ctx, vs.client, vmFolder, dummyVMPrefix)
+			for _, dummyVMRef := range dummyVMRefList {
+				deleteVM(ctx, dummyVMRef)
+			}
+		}
+		time.Sleep(CleanUpDummyVMRoutine_Interval * time.Minute)
+	}
+}
+
+// Get the dummy VM list from the kubernetes working directory.
+func getDummyVMList(ctx context.Context, c *govmomi.Client, vmFolder *object.Folder, dummyVMPrefix string) []*object.VirtualMachine {
+	vmFolders, _ := vmFolder.Children(ctx)
+	var vmFolderRefs []types.ManagedObjectReference
+	for _, vmFolder := range vmFolders {
+		vmFolderRefs = append(vmFolderRefs, vmFolder.Reference())
+	}
+
+	var vmRefs []types.ManagedObjectReference
+	for _, vmFolder := range vmFolderRefs {
```

Comment:
It seems to me that this for-loop can be easily merged with the previous one, right? And maybe also with the next one?